src/VisualStudio/Core/Def/Implementation/FindReferences/Contexts/AbstractTableDataSourceFindUsagesContext.cs

```diff
@@ -228,11 +222,11 @@ private void CancelSearch()
                 Presenter.AssertIsForeground();
 
                 // Cancel any in flight find work that is going on.
-                if (_cancellationTokenSource != null)
+                if (CancellationTokenSource != null)
                 {
-                    _cancellationTokenSource.Cancel();
-                    _cancellationTokenSource.Dispose();
-                    _cancellationTokenSource = null;
+                    CancellationTokenSource.Cancel();
+                    CancellationTokenSource.Dispose();
+                    CancellationTokenSource = null;
```

Comment:
I'm confused by the null-out here: the underlying data the CancellationTokenSource holds onto is gotten rid of in the Dispose() case, right? The docs say things like "you must release all references" but I can't tell if that's just automatic stuff the docs generate for all IDisposable things. Specifically this warning:

> Always call Dispose before you release your last reference to the CancellationTokenSource. Otherwise, the resources it is using will not be freed until the garbage collector calls the CancellationTokenSource object's Finalize method.

seems off because the implementation verbatim says:

> // There is nothing to do if disposing=false because the CancellationTokenSource holds no unmanaged resources.

@stephentoub is this null-out actually required, or is this just something the docs generate automatically for those who think IDisposable.Dispose() also makes the direct object go away?

Reply:
> Otherwise, the resources it is using will not be freed until the garbage collector calls the CancellationTokenSource object's Finalize method.

Honestly, if this is the case, I'd be 100% ok not disposing then at all. It just adds complexity to lots of scenarios that we could avoid if we know the GC will just handle this for us

Reply:
>  it is using will not be freed until the garbage collector calls the CancellationTokenSource object's Finalize method.

Hrmm... i took a look, and CTS doesn't even have a finalizer afaict... So i'm not sure if these comments are valid.